src/pages/transactions/[txid].tsx

```diff
@@ -1,52 +1,160 @@
+            {props.transaction.block.height}
+          </a>
+        </Link>
+      </AdaptiveList.Row>
+    </AdaptiveList>
+  )
+}
+
+function SummaryTableListRight (props: { transaction: Transaction, vins: TransactionVin[], vouts: TransactionVout[] }): JSX.Element {
+  const blockTime = format(fromUnixTime(props.transaction.block.medianTime), 'PPpp')
+
+  return (
+    <AdaptiveList className='w-full lg:w-1/2'>
+      <AdaptiveList.Row name='Fee Rate' testId='transaction-detail-fee-rate'>
+        {props.vins[0].vout === undefined ? 'No Inputs' : `${(getTransactionFee(props.transaction, props.vins).dividedBy(props.transaction.size)).decimalPlaces(8).toString()} mDFI/byte`}
+      </AdaptiveList.Row>
+      <AdaptiveList.Row name='Size' testId='transaction-detail-size'>
+        {props.transaction.size} bytes
+      </AdaptiveList.Row>
+      <AdaptiveList.Row name='Received Time' testId='transaction-detail-received-time'>
+        {blockTime}
+      </AdaptiveList.Row>
+    </AdaptiveList>
   )
 }
 
+function getTransactionFee (transaction: Transaction, vins: TransactionVin[]): BigNumber {
+  const fee = getTotalVinsValue(vins) - parseFloat(transaction.totalVoutValue)
+  return new BigNumber(fee).multipliedBy(1000)
+}
+
+function getTotalVinsValue (vins: TransactionVin[]): number {
+  let value = 0
+  vins.forEach(vin => {
+    if (vin.vout !== undefined) {
+      value += parseFloat(vin.vout.value)
+    }
+  })
+  return value
+}
+
 export async function getServerSideProps (context: GetServerSidePropsContext): Promise<GetServerSidePropsResult<TransactionPageProps>> {
   const api = getWhaleApiClient(context)
-  const network = context.query.network?.toString()
   const txid = context.params?.txid as string
 
-  if (network === 'MainNet' || network === undefined) {
-    return {
-      redirect: {
-        statusCode: 302,
-        destination: `https://mainnet.defichain.io/#/DFI/mainnet/tx/${txid}`
-      }
-    }
+  // Will improve with newer iteration of whale api
+  const vins: TransactionVin[] = []
+  let vinsResponse = await api.transactions.getVins(txid, 60)
+  vins.push(...vinsResponse)
+  while (vinsResponse.hasNext) {
+    vinsResponse = await api.transactions.getVins(txid, 60, vinsResponse.nextToken)
+    vins.push(...vinsResponse)
   }
 
-  if (network === 'TestNet') {
-    return {
-      redirect: {
-        statusCode: 302,
-        destination: `https://testnet.defichain.io/#/DFI/testnet/tx/${txid}`
-      }
-    }
+  const vouts: TransactionVout[] = []
+  let voutsResponse = await api.transactions.getVouts(txid, 60)
+  vouts.push(...voutsResponse)
+  while (voutsResponse.hasNext) {
+    voutsResponse = await api.transactions.getVouts(txid, 60, voutsResponse.nextToken)
+    vouts.push(...voutsResponse)
   }
 
   return {
     props: {
-      transaction: await api.transactions.get(txid)
+      transaction: await api.transactions.get(txid),
```

Comment:
when transaction not found we need to show an alternative page.

e.g. transaction txid not confirmed not found, it might not be confirmed yet ....

Reply:
![image](https://user-images.githubusercontent.com/4266087/135806153-907b39f8-2d40-4ca5-8e8d-b159d023b9a3.png)

something like this: https://mainnet.defichain.com/#/DFI/mainnet/tx/9693a839caeeb5c161bf5768d9b64cf7d7c1704f1e8e5caf3d9c08b18599ddba